src/persistence.js

```diff
@@ -39,51 +39,52 @@ export default function _Persistence(mpInstance) {
             if (!mpInstance._Store.isLocalStorageAvailable) {
                 mpInstance._Store.SDKConfig.useCookieStorage = true;
             }
-
-            // https://go.mparticle.com/work/SQDSDKS-6046
-            if (mpInstance._Store.isLocalStorageAvailable) {
-                storage = window.localStorage;
-                if (mpInstance._Store.SDKConfig.useCookieStorage) {
-                    // For migrating from localStorage to cookies -- If an instance switches from localStorage to cookies, then
-                    // no mParticle cookie exists yet and there is localStorage. Get the localStorage, set them to cookies, then delete the localStorage item.
-                    if (localStorageData) {
-                        if (cookies) {
-                            // https://go.mparticle.com/work/SQDSDKS-6047
-                            allData = mpInstance._Helpers.extend(
-                                false,
-                                localStorageData,
-                                cookies
-                            );
-                        } else {
-                            allData = localStorageData;
-                        }
-                        storage.removeItem(mpInstance._Store.storageName);
-                    } else if (cookies) {
-                        allData = cookies;
-                    }
-                    self.storeDataInMemory(allData);
-                } else {
-                    // For migrating from cookie to localStorage -- If an instance is newly switching from cookies to localStorage, then
-                    // no mParticle localStorage exists yet and there are cookies. Get the cookies, set them to localStorage, then delete the cookies.
-                    if (cookies) {
+            if (!mpInstance._Store.getNoFunctional()) {
```

Comment:
You can reduce the number of changes here by doing the inverse and returning early:


```suggestion
            if (mpInstance._Store.getNoFunctional()) {
                    return;
            }
            // then unindent everything you indented below
```

Reply:
I didn't do this as `initializeStorage` function is also used to initialize product storage and if i return early here then product storage will be affected